### ISSUE TITLE
By default, @src should not leak absolute path to the build directory

### ISSUE THREAD

**Issue by matklad** [ziglang/zig#16329]
```console
[matklad@Moby:~/p/zig]$ zig version
0.10.1

[matklad@Moby:~/p/zig]$ cat main.zig 
pub fn main() void {
    @import("std").debug.print("{s}", .{@src().file});
}

[matklad@Moby:~/p/zig]$ zig run main.zig 
/home/matklad/p/zig/main.zig
```

As you can see, `@src()` embedds absolute path to the source file into the executable. This is a poor default behavior, for two reasons: 

- this breaks reproducible builds --- building the program on a different machine will probably yield a different path
- it discloses my username (`matklad`) which I might want to keep private

It's unclear how to fix this. An obvious solution would be for src to include a relative path, anchored to the package root. But then we'll need a way to somehow identify which package we are talking about.

**Comment by nektro** [comment]
could be relative to the build root, so that it would be the same path that is shown in stack traces

**Comment by matklad** [comment]
Seems fixed in 0.11 already? 

```
19:05:23|~/p/tb/main|matklad/tmp-tigerbeetle⚡*?
λ bat main.zig
pub fn main() void {
    @import("std").debug.print("{s}", .{@src().file});
}

19:05:29|~/p/tb/main|matklad/tmp-tigerbeetle⚡*?
λ nix run nixpkgs#zig_0_10 run main.zig
/home/matklad/p/tb/main/main.zig⏎                                                                                                                                                                                                                                                                                                                                                              
19:05:31|~/p/tb/main|matklad/tmp-tigerbeetle⚡*?
λ nix run nixpkgs#zig_0_11 run main.zig
main.zig⏎       
```

Now I need to fix my hacky code which was relying on that :D

**Comment by matklad** [comment]
Optimisitcally closing.
